src/import-cycles.test.ts

```diff
@@ -0,0 +1,66 @@
+import assert from "assert"
+import { describe, it } from "mocha"
+import * as path from "path"
+import { detectImportCycles } from "./import-cycles"
+
+function examplePath(subPath: string) {
+	return path.join(__dirname, "../examples", subPath)
+}
+
+describe("Multiple cycle", async function () {
+	const importCycles = await detectImportCycles([
+		examplePath("multiple-cycles/entry.ts"),
+		examplePath("multiple-cycles/entry2.ts"),
+	])
+	it("should have 2 files that have cycles", async function () {
+		assert.equal(importCycles.length, 2)
+	})
+	it("entry.ts should have 3 cycles", async function () {
+		assert.equal(importCycles[0].cycle.length, 3)
+	})
+	it("entry2.ts should have 1 cycles", async function () {
+		assert.equal(importCycles[1].cycle.length, 1)
+	})
+})
```

Comment:
These tests are really hard to follow. I have no idea what cycles we should be seeing or not...

I think this can be improved by:
1. write the source code for each file directly in this test. write them to a tmp folder to execute the test. 
2. Write some comments, etc., that explain what the intention is for each line in the files that we're creating. 
3. It looks like we have multiple assertions here that might be better broken up into more isolated test examples. What about a single entry point and a single cycle? Simpler tests would be better.

Reply:
It's done feel free to let me know if you want me to do more test cases 🤠.